op-batcher/batcher/driver.go

```diff
@@ -209,6 +215,17 @@ func (l *BatchSubmitter) calculateL2BlockRangeToStore(ctx context.Context) (eth.
 		l.lastStoredBlock = syncStatus.SafeL2.ID()
 	}
 
+	// Ensure that the last stored block is still canonical. Not we need this check in addition to the internal check inside the batch
+	// queue because if there is a L2 reorg the L2 safe head on the sequence will be stuck until the batch submitter starts submitting again.
+	childCtx, cancel = context.WithTimeout(ctx, networkTimeout)
+	defer cancel()
+	if block, err := l.L2Client.BlockByNumber(childCtx, new(big.Int).SetUint64(l.lastStoredBlock.Number)); err != nil {
+		l.log.Info("Failed to check if the last stored block is canonical", "err", err)
```

Comment:
If this query fails, I suggest we `log.Warn` and also return the error directly.
```suggestion
	if block, err := l.L2Client.BlockByNumber(childCtx, new(big.Int).SetUint64(l.lastStoredBlock.Number)); err != nil {
		l.log.Warn("Failed to check if the last stored block is canonical", "err", err, "number", l.lastStoredBlock.Number)
		return eth.BlockID{}, eth.BlockID{}, fmt.Errorf("checking canonical last stored block: %w", err)
```
The next step is querying the blocks on the L2 client, which then probably also fails. Just returning an error here gives the L2 client time to come back online in the next loop tick.